### PR TITLE
FlightTask - Move ekf reset counter monitoring logic in the base FlightTask

### PR DESCRIPTION
**Pull request by bresch**
Until now, the FlightTask altitude was checking for the EKF quaternion reset, and the AutoLineSmoothVel & PositionSmoothVel were checking for position and velocity estimate resets; the code was simply duplicated.

To make it more scalable and to avoid further copy/paste, the logic for checking the reset counters has been moved to the base FlightTask that calls EKF reset handler functions.
Each child FlightTask can simply implement what it wants to do in case
of an EKF reset event by overriding one of the `_ekfResetHandler...` functions:
- void _ekfResetHandlerPositionXY()
- void _ekfResetHandlerVelocityXY()
- void _ekfResetHandlerPositionZ()
- void _ekfResetHandlerVelocityZ()
- void _ekfResetHandlerHeading(float delta_psi)

Changes made in
- [x] ManualAltitude
- [x] ManualPositionSmoothVel
- [x] AutoLineSmoothVel
- [x] AltitudeSmoothVel

Next step is to send the `delta_position` and `delta_velocity` as well, but this is more than a simple refactoring (see issue #12966)

### CODE REVIEW

**Comment by Stifael** [comment]
makes sense.

**Comment by Stifael** [comment]
one question:
Would it be possible to hide the reset from the child-tasks entirely? Basically the entire reset + delta-adjustment would be done in `FlightTasks`

**Comment by bresch** [comment]
@Stifael Unfortunately not. Some Flight Tasks, such as the ones using the "VelocitySmoothing" class need to adjust their internal state, not only the generated setpoint. However, we might want to have a basic reset handling in `FlightTask.cpp` and then just override it if required.

**Comment by bresch** [comment]
Rebased on master

File: src/lib/FlightTasks/tasks/ManualAltitudeSmoothVel/FlightTaskManualAltitudeSmoothVel.cpp [hunk 1]
```diff
@@ -60,8 +58,6 @@ void FlightTaskManualAltitudeSmoothVel::reActivate()
 	// The task is reacivated while the vehicle is on the ground. To detect takeoff in mc_pos_control_main properly
 	// using the generated jerk, reset the z derivatives to zero
 	_smoothing.reset(0.f, 0.f, _position(2));
-
-	_initEkfResetCounters();
```
Comment (MaEtUgR) [inline src/lib/FlightTasks/tasks/ManualAltitudeSmoothVel/FlightTaskManualAltitudeSmoothVel.cpp]: Is that case of `reActivate()` covered in the new implementation?
It's now "just" called in `activate()`

**Comment by MaEtUgR** [comment]
I pushed a suggestion for the struct initializer that IMHO is more readable but feel free to drop it if you don't agree.

File: src/lib/FlightTasks/tasks/ManualAltitudeSmoothVel/FlightTaskManualAltitudeSmoothVel.cpp (hunk 1 above)
Comment (bresch) [reply src/lib/FlightTasks/tasks/ManualAltitudeSmoothVel/FlightTaskManualAltitudeSmoothVel.cpp]: I don't see why we should re-initialize the counter during the reactivate; I know I put it there before, but I don't see why so I decided to remove it.

Comment (MaEtUgR) [reply src/lib/FlightTasks/tasks/ManualAltitudeSmoothVel/FlightTaskManualAltitudeSmoothVel.cpp]: true

**Pull request merged**